sdk/core/Azure.Core/src/DelayStrategy.cs

```diff
@@ -10,18 +10,25 @@ namespace Azure.Core
     /// <summary>
     /// Strategy to control delay behavior.
     /// </summary>
-    internal abstract class DelayStrategy
+    public abstract class DelayStrategy
     {
         /// <summary>
         /// Get the interval of next delay iteration.
         /// </summary>
         /// <remarks> Note that the value could change per call. </remarks>
         /// <param name="response"> Server response. </param>
+        /// <param name="attempt"></param>
         /// <param name="suggestedInterval"> Suggested pollingInterval. It is up to strategy
         /// implementation to decide how to deal with this parameter. </param>
         /// <returns> Delay interval of next iteration. </returns>
-        public abstract TimeSpan GetNextDelay(Response response, TimeSpan? suggestedInterval);
+        public abstract TimeSpan GetNextDelay(Response response, int attempt, TimeSpan? suggestedInterval);
```

Comment:
nit: should this be RetryNumber to match the public API on ProcessingContext?

Reply:
This would not be retry specific

Reply:
Would there ever be a scenario where we wanted delays in LRO polling to have a different strategy than retry delays in the same client?

Reply:
Yeah I think that is possible. We may want to call the options property `RetryStrategy` and potentially have a separate `LroStrategy`.

Reply:
For now I'm limiting the DelayStrategy to being exposed in RetryPolicy constructor. We can consider adding additional properties in ClientOptions for LRO and Retry to be used outside of a custom RetryPolicy.

Reply:
I've added an overload to `Operation<T>` based on @m-nash's suggestion. We could also consider a client option specific to LROs.

Reply:
I ended up changing this to RetryNumber as per @annelo-msft's initial suggestion - it felt odd to  use attempt number given that it would always start at 2 when the overrides are called.